ui/js/controllers/bugfiler.js

```diff
@@ -163,41 +165,91 @@ treeherder.controller('BugFilerCtrl', [
-                    failurePath = "testing/web-platform/tests/" + failurePath;
+                    if (failurePath.startsWith("mozilla/tests")) {
+                        failurePath = "testing/web-platform/" + failurePath;
+                    } else {
+                        failurePath = "testing/web-platform/tests/" + failurePath;
+                    }
                 }
 
                 // Search mercurial's moz.build metadata to find products/components
                 $scope.searching = "Mercurial";
-                $http.get("https://hg.mozilla.org/mozilla-central/json-mozbuildinfo?p=" + failurePath).then(function(request) {
-                    if (request.data.aggregate && request.data.aggregate.recommended_bug_component) {
-                        var suggested = request.data.aggregate.recommended_bug_component;
-                        $scope.suggestedProducts.push(suggested[0] + " :: " + suggested[1]);
+                $http.get("https://hg.mozilla.org/mozilla-central/json-mozbuildinfo?p=" + failurePath).then(function(firstRequest) {
+                    if (firstRequest.data.aggregate && firstRequest.data.aggregate.recommended_bug_component) {
+                        var suggested = firstRequest.data.aggregate.recommended_bug_component;
+                        addProduct(suggested[0] + " :: " + suggested[1]);
                     }
 
                     $scope.searching = false;
 
-                    if ($scope.suggestedProducts.length === 0) {
-                        var jg = selectedJob.job_group_name.toLowerCase();
-                        // Some job types are special, lets explicitly handle them.
-                        if (jg.includes("web platform")) {
-                            $scope.suggestedProducts.push("Testing :: web-platform-tests");
-                        }
-                        if (jg.includes("talos")) {
-                            $scope.suggestedProducts.push("Testing :: Talos");
-                        }
-                        if (jg.includes("mochitest") && failurePath.includes("webextensions/")) {
-                            $scope.suggestedProducts.push("Toolkit :: WebExtensions: General");
-                        }
-                        if (jg.includes("mochitest") && failurePath.includes("webrtc/")) {
-                            $scope.suggestedProducts.push("Core :: WebRTC");
-                        }
+                    // Make an attempt to find the file path via a dxr file search
+                    if ($scope.suggestedProducts.length === 0 && $uibModalInstance.possibleFilename.length > 4) {
+                        $scope.searching = "DXR & Mercurial";
+                        var dxrlink = "https://dxr.mozilla.org/mozilla-central/search?q=file:" + $uibModalInstance.possibleFilename + "&redirect=false&limit=5";
+                        $http.get(dxrlink, {"headers": {
+                            "Accept": "application/json"
```

Comment:
DXR seems like it takes either the "text/plain" or "/" options as a higher priority, because when I omit specifying the header, I see the request sending "application/json, text/plain, */*", but don't receive JSON in the response.